compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1551,13 +1555,18 @@ class Typer extends Namer
     if (sym.isInlineMethod) rhsCtx = rhsCtx.addMode(Mode.InlineableBody)
     val rhs1 = typedExpr(ddef.rhs, tpt1.tpe)(rhsCtx)
 
-    if (sym.isInlineMethod) PrepareInlineable.registerInlineInfo(sym, ddef.rhs, _ => rhs1)
+    if (sym.isInlineMethod) {
+      PrepareInlineable.checkInlineMacro(sym, rhs1, ddef.sourcePos)
+      PrepareInlineable.registerInlineInfo(sym, _ => rhs1)
+    }
 
     if (sym.isConstructor && !sym.isPrimaryConstructor)
       for (param <- tparams1 ::: vparamss1.flatten)
         checkRefsLegal(param, sym.owner, (name, sym) => sym.is(TypeParam), "secondary constructor")
 
-    assignType(cpy.DefDef(ddef)(name, tparams1, vparamss1, tpt1, rhs1), sym)
+    val ddef1 = assignType(cpy.DefDef(ddef)(name, tparams1, vparamss1, tpt1, rhs1), sym)
+    sym.defTree = ddef1
```

Comment:
I wonder whether we should factor out the `defTree` setting, into a method in `tpd.TreeOps` like:
```
def setDefTree: tree.type = {
  val sym = tree.symbol
  if (sym.exists) sym.defTree = tree
  tree
}
```
We seem to have 4 uses in this change set, and maybe more elsewhere.

Reply:
It seems worth it. I found 6 instances where it could be used. I will add a commit with this change.